### DISCUSSION TITLE
How to import custom models?

### DISCUSSION THREAD

**Discussion by ftt-prod** [Anjok07/ultimatevocalremovergui#805] [Q&A]
Hi, I have HP5_only_main_vocal.pth from RVC V2.
It works very well but i would like to use it in UVR.

I put it in the VR_Models folder, UVR detects it but asks me for the parameters, which i cannot find on the web, if someone can help me!

![image](https://github.com/Anjok07/ultimatevocalremovergui/assets/43067595/732cfc37-f360-4fc4-9540-6358b2cbb7af)

Which one should i use to this model?

**Comment by onasre** [comment]
Yah , did you get it to work?

**Comment by lbrooney** [comment]
Better late than never. To be clear this is only for `HP5_only_main_vocal` from RVC (Pretty sure the other models  from RVC do the same thing, but worth verifying for yourself)

I believe its `4band_v2`.  `HP5_only_main_vocal` is VR Architecture. Also the primary stem is instrumental.

Also if you want to get `HP5_only_main_vocal` to work in UVR(might break other models) you have to modify UVR:
In [`seperate.py`](https://github.com/Anjok07/ultimatevocalremovergui/blob/c975f04885a217aec8c06223c4b2508ca4956c6a/separate.py#L1036) (Base Folder) go to line 1036 and add `61968` to nn_arches, should look like:
`33966, 56817, 61968, 123821, 123812, 129605, 218409, 537238, 537227]`

In [`nets.py`](https://github.com/Anjok07/ultimatevocalremovergui/blob/c975f04885a217aec8c06223c4b2508ca4956c6a/lib_v5/vr_network/nets.py#L52) (`./lib_v5/vr_network/`) go to line 52 and add `61968` to hp_model_arch, should look like:
`hp_model_arch = [61968, 123821, 123812]`

Basically whats happening is that `HP5_only_main_vocal` is being recognized as a 5.1 model which it's not. This could potentially break other models, I don't know and I haven't tested, but keep it in mind. Can probably adapt this for other models.

Determining the param from RVC
[`uvr`](https://github.com/RVC-Project/Retrieval-based-Voice-Conversion-WebUI/blob/7728caad546c8d5dab17e724b41fbe9b28edaf30/infer-web.py#L1158C25-L1158C28) called from `inferweb`

Since [`HP5_only_main_vocal`](https://github.com/RVC-Project/Retrieval-based-Voice-Conversion-WebUI/blob/7728caad546c8d5dab17e724b41fbe9b28edaf30/infer/modules/uvr5/modules.py#L30) doesn't have `DeEcho` in its name its a `AudioPre`

[`AudioPre`](https://github.com/RVC-Project/Retrieval-based-Voice-Conversion-WebUI/blob/7728caad546c8d5dab17e724b41fbe9b28edaf30/infer/modules/uvr5/vr.py#L31C9-L31C85) sets the model parameters to `4band_v2`
